lib/app/app_router.dart

```diff
@@ -42,8 +43,20 @@ class AppRouter extends _$AppRouter {
           ],
           children: [
             AutoRoute(
-              page: PirateCoinsRoute.page,
-              path: "pirate-coins",
+              page: PirateCoinsTeacherRoute.page,
+              path: "pirate-coins-teacher",
+              children: [
+                AutoRoute(
+                  page: StatsRoute.page,
+                  path: "stats",
+                  title: (context, route) => "Stats",
+                ),
+              ],
+              title: (context, route) => "Pirate Coins",
+            ),
+            AutoRoute(
+              page: PirateCoinsStudentRoute.page,
+              path: "pirate-coins-student",
```

Comment:
It would be better for these to be nested and the difference in routes invisible, at least to students.